src/coreclr/src/vm/exceptionhandling.cpp

```diff
@@ -3948,6 +3952,10 @@ void ExceptionTracker::ResumeExecution(
     EH_LOG((LL_INFO100, "resuming execution at 0x%p\n", GetIP(pContextRecord)));
     EH_LOG((LL_INFO100, "^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n"));
 
+#if defined(HOST_OSX) && defined(HOST_ARM64)
```

Comment:
I think this might have been added to fix a case where we were trying to JIT a method which didn't exist (probably a dynamic method builder test).  Effectively the pre*stub threw an exception and the exception handler required us to return to managed code in the managed exception handler.  

So this too is introduced by the ad-hoc placement of the `PAL_JITWriteEnable(*)` calls.

I would suggest leaving this to be cleaned up by #41124.